SEED-platform/seed
# Add potable water meters

#### Any background context you want to provide?

#### What's this PR do?
- Adds water meter types: Potable Indoor, Potable Outdoor, Potable: Mixed Indoor/Outdoor
- The default display water unit is `kGal (thousand gallons) (US)`
- Provides unit conversions for the following water units to `kGal (thousand gallons) (US)`
```
    unit_conversion = {
        "ccf (hundred cubic feet)": 0.748,
        "cf (cubic feet)": 0.00748,
        "cGal (hundred gallons) (US)": 0.1,
        "cGal (hundred gallons) (UK)": 0.12,
        "cm (cubic meters)": 0.264172,
        "Gallons (US)": 0.001,
        "Gallons (UK)": 0.0012,
        "kcf (thousand cubic feet)": 7.48,
        "kcm (thousand cubit meters)": 264.172,
        "kGal (thousand gallons) (US)": 1.00,
        "kGal (thousand gallons) (UK)": 1.2,
        "Liters": 0.000264172,
        "Mcf (million cubic feed)": 7480.52,
        "MGal (million gallons) (UK)": 1000,
        "MGal (million gallons) (US)": 1200,
    }
```
NOTE: This does not include support for the other water types avaliable on Portfolio Manager: 
- Other: Mixed Indoor/Outdoor, Other Indoor, Other Outdoor, Reclaimed: Mixed Indoor/Outdoor, Reclaimed: Indoor, Reclaimed: Outdoor, Well Water: Mixed Indoor/Outdoor, Well Water: Indoor, Well Water: Outdoor

#### How should this be manually tested?
- Upload a file that contains potable water meter data (see JCC_ESPM_clean.xlsx)
- Upload its meter data
- Check inventory detail meters for a property wtih water meters
- Change water meter unit settings in organization settings > units
- Check inventory detail meters, meter readings and header should be updated for the chosen units.

#### What are the relevant tickets?
#4658

#### Screenshots (if appropriate)
Organization Settings > Units
![Screenshot 2024-07-19 at 10 08 21 AM](https://github.com/user-attachments/assets/5c487e47-cc46-47a9-a9a7-91b65d687f11)
![Screenshot 2024-07-19 at 10 08 25 AM](https://github.com/user-attachments/assets/41310af9-7b41-4f98-80cc-b5004eb47b72)
![Screenshot 2024-07-19 at 10 08 35 AM](https://github.com/user-attachments/assets/a9b677de-76c4-48d4-9a75-700f529593f3)

Inventory Detail Meters
![Screenshot 2024-07-19 at 9 42 22 AM](https://github.com/user-attachments/assets/395c467b-dcaa-4d96-a499-96c158aeacbf)

Meter Upload
![Screenshot 2024-07-19 at 9 26 06 AM](https://github.com/user-attachments/assets/1b7a03f6-626f-4b97-b9bf-a299fa124233)
